static/js/settings_panel_menu.js

```diff
@@ -115,7 +115,14 @@ export class SettingsPanelMenu {
             }
         }
 
-        this.$curr_li = this.li_for_section(section);
+        const $li_for_section = this.li_for_section(section);
+        if ($li_for_section.length === 0) {
+            // This happens when there is no such section or the user does not have
+            // permission to view that section.
+            return;
+        }
```

Comment:
This function doesn't currently have a case where it just returns without doing anything; why does doing so open the first section?

Reply:
It is confusing, also took me some time to understand. `activate_section_or_default` is basically called twice when overlay is opening, first from the `admin.launch` and then from the toggle callback (ie. `show_org_settings`) as `settings_toggle.highlight_toggle` is called from `admin.launch`. We basically do not set the `$curr_li` by returning early element such that, `current_tab` returns the first section when `show_org_settings`, otherwise `$curr_li` is set to non-existent jquery object and `current_tab` returns undefined.

Reply:
While writing this I have a better solution. We can do something like this where we just set `section` to the `current_tab` (first section when overlay is opened first time and the last open section in case overlay was opened already and then hash was changed to something invalid)-
```javascript
const $li_for_section = this.li_for_section(section);
if ($li_for_section.length === 0) {
        // This happens when there is no such section or the user does not have
        // permission to view that section.
        section = this.current_tab();
} else {
        this.$curr_li = $li_for_section;
}
```

Reply:
Can update the PR with above solution if this looks good,

Reply:
I don't totally understand but that sounds plausible. What does `current_tab()` do if we've never opened the settings UI?

Reply:
It returns the first sections since we initalize `$curr_li` field to first section in the constructor.